### ISSUE TITLE
Counting members that satisfy a condition within a group generates "cannot be converted to SQL." exception.

### ISSUE THREAD

**Issue by dimensional-difficulties** [linq2db/linq2db#699]
First of all, thank you again for all your hard work on this program.

I am trying to count the number of non-null members within a group using the following code:

```cs
var query = dataContext.TRMasters(tablePath)
.Where(table => table.CampaignID == campaignID && table.TouchID == touchID)
.GroupBy(table => table.sentTimestamp.Date)
.Select(ppp => new
{
    date = ppp.Key,
    openCount = ppp.Count(pppp => pppp.FirstOpenTimestamp != null),
    clickCount = ppp.Count(pppp => pppp.FirstClickTimestamp != null),
    sentCount = ppp.Count(),
    replyCount = ppp.Count(pppp => pppp.IndReply)
});
```

However, when this code runs, I get an error saying
"An exception of type 'LinqToDB.LinqToDBException' occurred in linq2db.dll but was not handled in user code

Additional information: 'pppp.FirstOpenTimestamp' cannot be converted to SQL."

**Comment by sdanyliv** [comment]
@dimensional-difficulties, it is related to #679 and #680.
We will extend aggregate functions, i hope, in one-two weeks.

**Comment by ili** [comment]
@dimensional-difficulties hi!
can you please provide your `ppp` object definition?

**Comment by dimensional-difficulties** [comment]
Here is a simplified version of the class with the definitions relevant to the query in question.

public class TRMaster
    {
        [Column(Name = CommonConst.CAMPAIGN_ID_COLUMN)]
        public string CampaignID { get; set; }

        [Column(Name = CommonConst.TOUCH_ID_COLUMN)]
        public string TouchID { get; set; }
        
        [Column(Name = MasterRollupConst.FIRST_CLICK_TIMESTAMP_COLUMN)]
        public DateTime FirstClickTimestamp { get; set; }

        [Column(Name = MasterRollupConst.FIRST_OPEN_TIMESTAMP_COLUMN)]
        public DateTime FirstOpenTimestamp { get; set; }

        [Column(Name = CommonConst.REPLY_INDICIATOR_COLUMN)]
        public bool IndReply { get; set; }

    }

**Comment by sdanyliv** [comment]
@ili, it is reproduced, but needs @igor-tkachev for correct fix.

**Comment by ili** [comment]
fixed via #775
